community/QClass_2024/Submissions/HW1/Christos_Koromilas_HW1_QClass2024.ipynb

```diff
@@ -0,0 +1 @@
y_controlled_ry(control_bit: QBit, target: QBit) -> None:\n","    control(control_bit, lambda: RY(pi / 2, target))\n","\n","\n","@qfunc\n","def main(control_bit: Output[QBit], target: Output[QBit]) -> None:\n","    allocate(1, control_bit)\n","    allocate(1, target)\n","    my_controlled_ry(control_bit, target)\n","\n","qmod = create_model(main)\n","qprog = synthesize(qmod)\n","show(qprog)"]},{"cell_type":"markdown","metadata":{"id":"ReKCKMQPAgo9"},"source":["### Exercise 5b - Control (\"Quantum If\")\n","The `control` operator is the conditional application of some operation, with the condition being that all control qubits are in the state |1>. This notion is generalized in QMOD to other control states, where the condition is specified as a comparison between a quantum numeric variable and a numeric value, similar to a classical `if` statement. Quantum numeric variables are declared with class `QNum`.\n","\n","See also [Numeric types](https://docs.classiq.io/latest/user-guide/platform/qmod/language-reference/quantum-types/#syntax).\n","\n","In QMOD this generalization is available as a native statement - control.\n","\n","See also [control](https://docs.classiq.io/latest/user-guide/platform/qmod/language-reference/statements/quantum-operators/).\n","\n","1. Declare a `QNum` output argument using `Output[QNum]` and name it `x`.\n","2. Use the `prepare_int` function to initialize it to `9`. Note that you don't need to specify the `QNum` attributes - size, sign, and fraction digits, as they are inferred at the point of initialization.\n","3. Execute the circuit and observe the results.\n","4. Declare another output argument of type `QBit` and perform a `control` such that under the condition that `x` is 9, the qubit is flipped. Execute the circuit and observe the results. Repeat for a different condition."]},{"cell_type":"code","execution_count":13,"metadata":{"id":"tISW82ILAgo9","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1716212713430,"user_tz":-120,"elapsed":7382,"user":{"displayName":"Christos Koromilas","userId":"17599903479737751299"}},"outputId":"e23bf946-2b7e-4511-ec24-a380004da7ad"},"outputs":[{"output_type":"stream","name":"stdout","text":["Opening: https://platform.classiq.io/circuit/b6600985-72b5-46b8-a02d-bff53d677218?version=0.41.1\n"]}],"source":["from classiq import *\n","\n","@qfunc\n","def main(x: Output[QNum], target: Output[QBit]) -> None:\n","    prepare_int(9, x)\n","    allocate(1, target)\n","    control(x == 9, lambda: X(target))\n","\n","qmod = create_model(main)\n","qprog = synthesize(qmod)\n","show(qprog)"]}],"metadata":{"kernelspec":{"display_name":"Python 3 (ipykernel)","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.11.8"},"colab":{"provenance":[]}},"nbformat":4,"nbformat_minor":0}
```

Comment:
<p>Hello @<em>Infi-DOS !</em></p><p>The notebook seems to be failing our test. </p><p>Can you please delete those broken links or replace them with this updated one: <a href="https://docs.classiq.io/latest" target="_blank">https://docs.classiq.io/latest/</a> ?</p><p>My apologies for that. </p><p>Thanks and well done for the great job!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/152/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>